recipes/crypto-createcipheriv-migration/src/workflow.ts

```diff
@@ -0,0 +1,428 @@
+	}
+
+	for (const [statement, change] of statementChanges) {
+		const edit = applyStatementChanges(statement, change);
+		if (edit) edits.push(edit);
+	}
+
+	if (edits.length === 0) return null;
+
+	return rootNode.commitEdits(edits);
+}
+
+function collectCallEdits({
+	rootNode,
+	statement,
+	binding,
+	kind,
+	edits,
+	statementChanges,
+	seenCallRanges,
+}: CollectParams) {
+	const patterns = [
+		`${binding}($ALGORITHM, $PASSWORD, $OPTIONS)`,
+		`${binding}($ALGORITHM, $PASSWORD)`,
+	];
+
+	const calls = rootNode.findAll({
+		rule: {
+			any: patterns.map((pattern) => ({ pattern })),
+			kind: 'call_expression',
+		},
+	});
+
+	for (const call of calls) {
+		const rangeKey = getRangeKey(call);
+		if (seenCallRanges.has(rangeKey)) continue;
+		seenCallRanges.add(rangeKey);
+
+		const algorithmNode = call.getMatch('ALGORITHM');
+		const passwordNode = call.getMatch('PASSWORD');
+
+		if (!algorithmNode || !passwordNode) continue;
+
+		const algorithm = algorithmNode.text().trim();
+		const password = passwordNode.text().trim();
+		if (!algorithm || !password) continue;
+
+		const optionsText = call.getMatch('OPTIONS')?.text()?.trim();
+
+		const replacement =
+			kind === 'cipher'
+				? buildCipherReplacement({
+						binding,
+						algorithm,
+						password,
+						options: optionsText,
+					})
+				: buildDecipherReplacement({
+						binding,
+						algorithm,
+						password,
+						options: optionsText,
+					});
+
+		edits.push(call.replace(replacement));
+
+		if (isDestructuredStatement(statement)) {
+			const change = ensureStatementChange(statementChanges, statement);
+			// Ensure the binding points to the iv-based API
+			const sourceName = kind === 'cipher' ? 'createCipher' : 'createDecipher';
+			const targetName = `${sourceName}iv`;
+			change.rename.set(sourceName, targetName);
+			if (kind === 'cipher') {
+				change.additions.add('randomBytes');
+			}
+			change.additions.add('scryptSync');
+		}
+	}
+}
+
+function buildCipherReplacement(params: {
+	binding: string;
+	algorithm: string;
+	password: string;
+	options?: string;
+}): string {
+	const { binding, algorithm, password, options } = params;
+	const randomBytesCall = getMemberAccess(binding, 'randomBytes');
+	const scryptCall = getMemberAccess(binding, 'scryptSync');
+	const cipherCall = getCallableBinding(binding, 'createCipheriv');
+
+	const lines = [
+		'(() => {',
+		`\tconst __dep0106Salt = ${randomBytesCall}(16);`,
+		'\tconst __dep0106Key = ' +
+			scryptCall +
+			'(' +
+			password +
+			', __dep0106Salt, 32);',
+		`\tconst __dep0106Iv = ${randomBytesCall}(16);`,
+		'\t// DEP0106: Persist __dep0106Salt and __dep0106Iv with the ciphertext so it can be decrypted later.',
+		'\t// DEP0106: Adjust the derived key length (32 bytes) and IV length to match the chosen algorithm.',
+		'\treturn ' +
+			cipherCall +
+			'(' +
+			algorithm +
+			', __dep0106Key, __dep0106Iv' +
+			(options ? `, ${options}` : '') +
+			');',
+		'})()',
+	];
+
+	return lines.join(EOL);
```

Comment:
A few notes:

1. Just use `\n`. The system-specific EOL isn't _really_ needed here, and `\n` is standard.
2. Let's not use an Array + join. I'd rather store this as a template.js file, and do `.replace("__VARIABLE__", VALUE)` or something